scripts/data_ingest/main.py

```diff
@@ -0,0 +1,147 @@
+import logging
+import logging.config
+import os
+import sys
+from http import HTTPStatus
+from pathlib import Path
+from pprint import pprint
+
+import requests
+
+ADMIN_EMAIL_ENV = "SUPERUSER_EMAIL"
+ADMIN_PASSWORD_ENV = "SUPERUSER_PASSWORD"
+ADMIN_TOKEN_ENV = "SUPERUSER_TOKEN"
+BULK_IMPORT_ENDPOINT = "/api/v1/bulk-imports/cclw/law-policy"
+
+
+DEFAULT_LOGGING = {
+    "version": 1,
+    "disable_existing_loggers": False,
+    "formatters": {
+        "standard": {"format": "%(asctime)s [%(levelname)s] %(name)s: %(message)s"},
+    },
+    "handlers": {
+        "default": {
+            "level": "INFO",
+            "formatter": "standard",
+            "class": "logging.StreamHandler",
+            "stream": "ext://sys.stdout",  # Default is stderr
+        },
+    },
+    "loggers": {
+        "": {  # root logger
+            "handlers": ["default"],
+            "level": "INFO",
+        },
+        "__main__": {  # if __name__ == '__main__'
+            "handlers": ["default"],
+            "level": "DEBUG",
+            "propagate": False,
+        },
+    },
+}
+
+logging.config.dictConfig(DEFAULT_LOGGING)
+logger = logging.getLogger(__file__)
+
+
+def _log_response(response: requests.Response) -> None:
+    if response.status_code >= 400:
+        logger.error(
+            f"There was an error during a request to {response.url}. "
+            f"STATUS: {response.status_code}, BODY:{response.content!r}"
+        )
+
+    logger.debug(f"STATUS: {response.status_code}, BODY:{response.content!r}")
+
+
+def get_admin_token() -> str:
+    """Go through the login flow & create access token for requests."""
+    admin_user = os.getenv(ADMIN_EMAIL_ENV)
+    admin_password = os.getenv(ADMIN_PASSWORD_ENV)
+
+    if admin_user is None or admin_password is None:
+        raise RuntimeError("Admin username & password env vars must be set")
+
+    response = requests.post(
+        get_request_url("/api/tokens"),
+        headers={"Content-Type": "application/x-www-form-urlencoded"},
+        data={"username": admin_user, "password": admin_password},
+    )
+    _log_response(response=response)
+
+    token: str = response.json()["access_token"]
+    return token
+
+
+def get_admin_auth_headers() -> dict[str, str]:
+    """Create the required auth headers for requests."""
+    if (admin_user_token := os.getenv(ADMIN_TOKEN_ENV)) is None:
+        admin_user_token = get_admin_token()
+        os.environ[ADMIN_TOKEN_ENV] = admin_user_token
+
+    return {
+        "Authorization": "Bearer {}".format(admin_user_token),
+        "Accept": "application/json",
+    }
+
+
+def get_request_url(endpoint: str) -> str:
+    api_host = os.getenv("API_HOST", "http://backend:8888").rstrip("/")
```

Comment:
Function comments for this and post_data_ingest functions?

Reply:
I really should have written a get out of jail clause into those coding standards :joy: